realm/realm-library/src/main/java/io/realm/RealmResults.java

```diff
@@ -149,6 +152,13 @@ public boolean isValid() {
      */
     public RealmQuery<E> where() {
         realm.checkIfValid();
+        try {
```

Comment:
* Tricky part. Now we just return a `null` to user. And in the real world I can predict there will be some user met the `NullPointerException` because of this.
* Another option is we throw a checked exception to user to force user to catch it. But we don't use any checked exception in the lib, it might look like strange we have one here.
* IllegalStateException is something i don't like in this case, since there are some other case could also throw ISE here. eg.: a closed Realm ----- which is a user's bug should be fixed.